File: ViewModels/Pages/KeyDetectionFormatting.vb
Imports System.Globalization

Namespace ViewModels.Pages
    Friend Module KeyDetectionFormatting
        Public Function FormatCount(value As Integer) As String
            Return value.ToString(CultureInfo.InvariantCulture)
        End Function

        Public Function FormatMilliseconds(value As Nullable(Of Double)) As String
            If Not value.HasValue OrElse Double.IsNaN(value.Value) OrElse Double.IsInfinity(value.Value) Then
                Return "--"
            End If

            If Math.Abs(value.Value) < 10.0 Then
                Return value.Value.ToString("0.00", CultureInfo.InvariantCulture) & " ms"
            End If

            Return value.Value.ToString("0.0", CultureInfo.InvariantCulture) & " ms"
        End Function
    End Module
End Namespace

File: src/ClickSyncMouseTester/ViewModels/Pages/KeyDetectionFormatting.cs
using System;
using System.Globalization;

namespace ClickSyncMouseTester.ViewModels.Pages;

internal sealed class KeyDetectionFormatting
{
    public static string FormatCount(int value)
    {
        return value.ToString(CultureInfo.InvariantCulture);
    }

    public static string FormatMilliseconds(double? value)
    {
        if (!value.HasValue || double.IsNaN(value.Value) || double.IsInfinity(value.Value))
        {
            return "--";
        }
        if (Math.Abs(value.Value) < 10.0)
        {
            return value.Value.ToString("0.00", CultureInfo.InvariantCulture) + " ms";
        }
        return value.Value.ToString("0.0", CultureInfo.InvariantCulture) + " ms";
    }
}





